contracts/deployment_scripts/testnet/001_deploy_tokens.ts

```diff
@@ -0,0 +1,40 @@
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import {DeployFunction} from 'hardhat-deploy/types';
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+    const { 
+        deployments, 
+        getNamedAccounts
+    } = hre;
+
+    const {deployer} = await getNamedAccounts();
+
+    await deployments.deploy('HOCERC20', {
```

Comment:
Mainly that the plugin persists what it has deployed, makes it accessible by name in other deployments and knows on which network it is deployed so it can manage complicated bits. Also it has some fail safes to prevent deploying when stuff is out of order.

Reply:
In the example you gave, hardhat when started instantiates some singleton stuff to the network it was started with and caches a ton of stuff so `hre.ethers` always points to a single network. We'd have to manually instantiate stuff when we want to orchestrate cross network deployments. But yeah, we can do it entirely through ethers, it would just be additional code and manually managing what account is connected and so on.